snmp.yml

```diff
@@ -1221,3 +1221,377 @@ apcups:
+      oid: 1.3.6.1.2.1.15.3.1.19
+      indexes:
+        - labelname: bgpPeerIdentifier
+          type: IpAddress
+      lookups:
+        - labels: [bgpPeerIdentifier]
+          labelname: bgpPeerLocalAddr
+          oid: 1.3.6.1.2.1.15.3.1.5
+        - labels: [bgpPeerIdentifier]
+          labelname: bgpPeerRemoteAddr
+          oid: 1.3.6.1.2.1.15.3.1.7
+    - name: bgpPeerHoldTimeConfigured
+      oid: 1.3.6.1.2.1.15.3.1.20
+      indexes:
+        - labelname: bgpPeerIdentifier
+          type: IpAddress
+      lookups:
+        - labels: [bgpPeerIdentifier]
+          labelname: bgpPeerLocalAddr
+          oid: 1.3.6.1.2.1.15.3.1.5
+        - labels: [bgpPeerIdentifier]
+          labelname: bgpPeerRemoteAddr
+          oid: 1.3.6.1.2.1.15.3.1.7
+    - name: bgpPeerKeepAliveConfigured
+      oid: 1.3.6.1.2.1.15.3.1.21
+      indexes:
+        - labelname: bgpPeerIdentifier
+          type: IpAddress
+      lookups:
+        - labels: [bgpPeerIdentifier]
+          labelname: bgpPeerLocalAddr
+          oid: 1.3.6.1.2.1.15.3.1.5
+        - labels: [bgpPeerIdentifier]
+          labelname: bgpPeerRemoteAddr
+          oid: 1.3.6.1.2.1.15.3.1.7
+    - name: bgpPeerMinASOriginationInterval
+      oid: 1.3.6.1.2.1.15.3.1.22
+      indexes:
+        - labelname: bgpPeerIdentifier
+          type: IpAddress
+      lookups:
+        - labels: [bgpPeerIdentifier]
+          labelname: bgpPeerLocalAddr
+          oid: 1.3.6.1.2.1.15.3.1.5
+        - labels: [bgpPeerIdentifier]
+          labelname: bgpPeerRemoteAddr
+          oid: 1.3.6.1.2.1.15.3.1.7
+    - name: bgpPeerMinRouteAdvertisementInterval
+      oid: 1.3.6.1.2.1.15.3.1.23
+      indexes:
+        - labelname: bgpPeerIdentifier
+          type: IpAddress
+      lookups:
+        - labels: [bgpPeerIdentifier]
+          labelname: bgpPeerLocalAddr
+          oid: 1.3.6.1.2.1.15.3.1.5
+        - labels: [bgpPeerIdentifier]
+          labelname: bgpPeerRemoteAddr
+          oid: 1.3.6.1.2.1.15.3.1.7
+    - name: bgpPeerInUpdateElapsedTime
+      oid: 1.3.6.1.2.1.15.3.1.24
+      indexes:
+        - labelname: bgpPeerIdentifier
+          type: IpAddress
+      lookups:
+        - labels: [bgpPeerIdentifier]
+          labelname: bgpPeerLocalAddr
+          oid: 1.3.6.1.2.1.15.3.1.5
+        - labels: [bgpPeerIdentifier]
+          labelname: bgpPeerRemoteAddr
+          oid: 1.3.6.1.2.1.15.3.1.7
+
+# The Cisco BGP4 MIB is proprietary to Cisco routers. Not all routers
+# support all parts of the MIB, even if they are running BGP so your
+# mileage may vary. If possible, upgrade to the latest version of the
+# firmware possible.
+#
+# ftp://ftp.cisco.com/pub/mibs/v2/CISCO-BGP4-MIB.my
+
+cbgp4:
+  version: 2
+  walk:
+    - 1.3.6.1.4.1.9.9.187.1.2
+  metrics:
+    - name: cbgpPeerAcceptedPrefixes
+      oid: 1.3.6.1.4.1.9.9.187.1.2.4.1.1
+      indexes:
+        - labelname: bgpPeerRemoteAddr
+          type: IpAddress
+        - labelname: cbgpPeerAddrFamilyAfi
+          type: Integer32
```

Comment:
I can see your point about "ipv4" being more meaningful. There are potentially a lot of textual conventions that simply map an integer to a label though. Adding code for each one I think would be a maintenance issue. I think that it'd be better to have a generic mechanism in the configuration for defining these, especially if they could eventually be auto-generated. Also, since the indexes are used in looking up other OIDs, converting them from integers to string labels would mean a lot more complexity in making matches.